### ISSUE TITLE
Default recursive doesn't work

### ISSUE THREAD

**Issue by ZsoltPath** [LanceMcCarthy/Action-AzureBlobUpload#105]
Hi,

Sometimes it doesn't upload subdirectories. See log snippet.
The list of the files created by a `find dist` command.

```

2021-01-29T12:21:41.6825240Z [12:21:41] Using gulpfile ~/work/kundkhu/kundkhu/gulpfile.js
2021-01-29T12:21:41.6836695Z [12:21:41] Starting 'build'...
2021-01-29T12:21:41.6850326Z [12:21:41] Starting 'nunjucks'...
2021-01-29T12:21:41.6852331Z Rendering nunjucks files..
2021-01-29T12:21:41.7522048Z [12:21:41] Finished 'nunjucks' after 67 ms
2021-01-29T12:21:41.7527514Z [12:21:41] Finished 'build' after 69 ms
2021-01-29T12:21:41.7649242Z dist
2021-01-29T12:21:41.7650816Z dist/index.html
2021-01-29T12:21:41.7653647Z dist/szakacskonyv
2021-01-29T12:21:41.7655308Z dist/szakacskonyv/original
2021-01-29T12:21:41.7656803Z dist/szakacskonyv/original/122.jpg
2021-01-29T12:21:41.7658262Z dist/szakacskonyv/original/55.jpg
2021-01-29T12:21:41.7659585Z dist/szakacskonyv/original/59.jpg
2021-01-29T12:21:41.7732347Z dist/szakacskonyv/eszt_rost.html
2021-01-29T12:21:41.7732990Z dist/szakacskonyv/bufsztek.html
2021-01-29T12:21:41.7733587Z dist/szakacskonyv/images
2021-01-29T12:21:41.7734165Z dist/szakacskonyv/images/122.jpg
2021-01-29T12:21:41.7734764Z dist/szakacskonyv/images/55.jpg
2021-01-29T12:21:41.7735900Z dist/szakacskonyv/images/96-2.jpg
2021-01-29T12:21:41.7736508Z dist/szakacskonyv/images/59.jpg
2021-01-29T12:21:41.7850478Z dist/szakacskonyv/tolt_rost.html
2021-01-29T12:21:41.7851003Z dist/szakacskonyv/tejf.html
2021-01-29T12:21:41.7851528Z dist/szakacskonyv/vadnyul.html
2021-01-29T12:21:41.7852085Z dist/szakacskonyv/sonkas_tek.html
2021-01-29T12:21:41.7852933Z dist/szakacskonyv/csaszarszelet.html
2021-01-29T12:21:41.7853681Z dist/szakacskonyv/szakacskonyv copy.html
2021-01-29T12:21:41.7854668Z dist/szakacskonyv/nyulpastetom.html
2021-01-29T12:21:41.7855389Z dist/szakacskonyv/szakacskonyv.html
2021-01-29T12:21:41.7923169Z ##[group]Run LanceMcCarthy/Action-AzureBlobUpload@v1.8.0
2021-01-29T12:21:41.7923732Z with:
2021-01-29T12:21:41.7926342Z   connection_string: ***
2021-01-29T12:21:41.7926879Z   container_name: $web
2021-01-29T12:21:41.7927250Z   source_folder: dist/
2021-01-29T12:21:41.7927852Z   fail_if_source_empty: true
2021-01-29T12:21:41.7928497Z   clean_destination_folder: false
2021-01-29T12:21:41.7928968Z ##[endgroup]
2021-01-29T12:21:42.5629647Z Uploaded dist/index.html to $web/index.html...
```

**Comment by LanceMcCarthy** [comment]
Does it work if you downgrade to 1.7.1?

Im investigating a possible regression in the folder iteration logic.

**Comment by ZsoltPath** [comment]
@LanceMcCarthy, haven't checked.
Created https://github.com/LanceMcCarthy/Action-AzureBlobUpload/pull/106

**Comment by LanceMcCarthy** [comment]
I've published `v1.8.1`, let me know if you have any further issues.
